masterplan_tools/method/publicspace/ps_greedy.py

```diff
@@ -0,0 +1,612 @@
+                            if self.graph.nodes[neigh][func[closed_func]] == 1:
+                                flag = False
+                                break
+                        if flag:
+                            ls -= 1
+                            clsd[n] -= 1
+                            self.graph.nodes[n]["type"] = "потенциальное"
+                            self.graph.nodes[n]["functional_type"] = ""
+                            for neigh in neighbors:
+                                self.graph.nodes[neigh][func[closed_func]] -= 1
+                        flag = True
+                if ls >= zones_open:
+                    for k in clsd:
+                        if clsd[k] == -1:
+                            self.graph.nodes[k]["type"] = "существующее"
+                            self.graph.nodes[k]["functional_type"] = closed_func
+                            neighbor = list(self.graph.neighbors(k))
+                            for neigh in neighbor:
+                                self.graph.nodes[neigh][func[closed_func]] += 1
+                            clsd[k] = 0
+                        elif clsd[k] == 1:
+                            self.graph.nodes[k]["functional_type"] = ""
+                            self.graph.nodes[k]["type"] = "потенциальное"
+                            neighbors = list(self.graph.neighbors(k))
+                            for neigh in neighbors:
+                                self.graph.nodes[neigh][func[closed_func]] -= 1
+                            clsd[k] = 0
+                    ls = zones_open
+                    i += 1
+                else:
+                    i = 0
+                    nodes = list(self.graph)
+                    zones_open = ls
+                    clsd = {n: 0 for n in self.graph}
+            else:
+                i += 1
+
+    def provision(self):
+        for n in self.graph:
+            if self.graph.nodes[n]["is_zone"] == False:
+                self.graph.nodes[n]["provision"] = 0
+                if self.graph.nodes[n]["provision_mf"] > 0:
+                    self.graph.nodes[n]["provision"] += 0.25
+                if self.graph.nodes[n]["provision_ev"] > 0:
+                    self.graph.nodes[n]["provision"] += 0.25
+                if self.graph.nodes[n]["provision_ed"] > 0:
+                    self.graph.nodes[n]["provision"] += 0.25
+                if self.graph.nodes[n]["provision_sp"] > 0:
+                    self.graph.nodes[n]["provision"] += 0.25
+
+    def get_ratio(self, func_type, area):
+        r = [[0.53, 0.05, 0.1, 0.17, 0.15], [0.6, 0.1, 0.1, 0.2], [0.67, 0.1, 0.23], [0.6, 0.1, 0.2, 0.1]]
+        if func_type == "Многофункциональное":
+            a = f"Отдых и общение: {round(r[0][0] * area, 2)} м^2, Обслуживание базовых потребностей: {round(r[0][1] * area, 2)} м^2, Событийная: {round(r[0][2] * area, 2)} м^2, Спортивная: {round(r[0][3] * area, 2)} м^2, Учебная: {round(r[0][4] * area, 2)} м^2"

```

Comment:
1. Please stop using variable name `a`.
2. You can split really long f-strings just like the ordinary strings:
  ```python
string_result = (
    f"hello {'w'}"
    "orld"
    f" of f-strings {3-2}"
)```